lib/spack/llnl/util/filesystem.py

```diff
@@ -472,3 +474,110 @@ def find_library_path(libname, *paths):
         if os.path.exists(library):
             return path
     return None
+
+# Utilities for libraries
+
+
+class LibraryList(collections.Sequence):
+    """Manages a list of absolute paths to libraries"""
+    def __init__(self, libraries):
+        self.libraries = list(libraries)
+
+    @property
+    def directories(self):
```

Comment:
donno what's the convention, but maybe one-line doc-string for properties? Especially for things like `libnames` vs `names`.